contacts.py

```diff
@@ -584,13 +571,38 @@ def sync(self, data, args):
             since_string = "1970-01-01T01:01Z"
 
         since = unix_time_from_iso(since_string)
-        contacts, latest_contact_time = self.contact_dict.get_file_paths_between_times(since, now)
-        locations, latest_location_time = self.spatial_dict.get_file_paths_between_times(since, now)
-        latest_time = max(latest_contact_time, latest_location_time)
-
-        ret = {'until': iso_time_from_seconds_since_epoch(latest_time),
-               'since': since_string}
 
+        # correlate the two dictionaries
+        # ist of (timecode, serial_number, listL_type) between since and until
+        data = sortedlist(key = lambda k: k[0])
+        for the_dict in self.map_over_dicts():
+            current_list = the_dict.sorted_list_by_time_and_serial_number
+            data.update(map(lambda item: (item[0], item[1], the_dict), current_list[current_list.bisect_left((since, 0)):current_list.bisect_left((now, 0))]))
+            
+        length = len(data)
+        number_to_return = int(self.config.get('MAX_SYNC_COUNT', 1000))
```

Comment:
Doesn't config have a function getint ?

Reply:
i like treating it as a regular dict, just in case we change it.